### DISCUSSION TITLE
Using Base framework, but got Error: 'NoneType' Object  has no attribute 'forecast'

### DISCUSSION THREAD

**Discussion by AlvinAi96** [PaddlePaddle/PaddleSpatial#97] [BaiduKDDCup2022] [answered]
My Team Name: Dynamo
My User Name: Alvin9603

In 2022-05-23 17:19, I succeeded in submitting my  code and got the LB score. But after that time, I submit my code and keep failing, which report error: **'NoneType' Object  has no attribute 'forecast'**. I used Baidu evaluation.py to offline debug and it could be run successfully. As the below code, I can work with ``forecast_module = Loader.load(envs["pred_file"]) ``on my own but I don't know why it failed on the online submission. Would you like to take a look about it? I will be appreciate it.
My latest submission is in the 2022-05-24 14:51.
![image](https://user-images.githubusercontent.com/42187008/169969854-357a1885-f2f9-4112-85ae-0e297bea430e.png)

**Comment by AlvinAi96** [comment, marked as answer]
Hi, @ramber1836,

I guess the error is caused by the predict.forecast(). When the Loader() load the predict module, it failed and then return None. But I run the evaluation.py with zip file on my own, no error is reported. Is possible some libraries are unavailable under the base framework. If so, i suggest the LB log could print out which the library is lacked. 

Looking forward to check the backend log. Wating for your reply. Thanks! 

BTW, could you install lightgbm and xgboost on other frameworks, which we could do the model ensemble. If possible, it would be better if we could install library as we like.
 
![image](https://user-images.githubusercontent.com/42187008/169971333-ccf644f3-ff0f-44c0-8a93-d0ee17e45bd0.png)

**Reply by AlvinAi96** [reply]
I pre-downloaded the lightgbm library and packaged to upload it. The error is fixed.

**Reply by ramber1836** [reply]
@AlvinAi96 The lightgbm has been installed in different envs. Please check it.

**Comment by ramber1836** [comment]
You can check it again, the 'lightgbm not found' error has been fixed in the base environment.

**Reply by zlj-cs** [reply]
'No module named lightgbm' appeared just now when I completed my submission

**Reply by zlj-cs** [reply]
> 
![image](https://user-images.githubusercontent.com/65325751/170337218-c64fe977-068d-4145-8a61-538b5df65d7b.png)
I uploaded my submission, and then this error occurred

**Reply by LZY-the-boys** [reply]
![image](https://user-images.githubusercontent.com/72137647/170418707-ff617674-ef85-4a13-8e74-e9c97321316d.png)
me too

**Reply by wang975679801** [reply]
me too

**Reply by ramber1836** [reply]
'lightgbm not found' error is fixed now.
